### PR TITLE
Add MongoDB CRD

### PR DESCRIPTION
**Pull request by tamalsaha**

### CODE REVIEW

File: apis/kubedb/mongodb_types.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package kubedb
+
+import (
+	"github.com/appscode/go/encoding/json/types"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ResourceCodeMongoDB = "mn"
```
Comment (aerokite) [inline apis/kubedb/mongodb_types.go]: May be we can use "mdb".

File: apis/kubedb/v1alpha1/mongodb_types.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	Reason       string        `json:"reason,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MongoDBList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// Items is a list of MongoDB TPR objects
+	Items []*MongoDB `json:"items,omitempty"`
```
Comment (aerokite) [inline apis/kubedb/v1alpha1/mongodb_types.go]: Do not use pointer in List Items

File: apis/kubedb/v1alpha1/mysql_types.go [hunk 3]
```diff
@@ -25,9 +25,9 @@ type MySQL struct {
 }
 
 type MySQLSpec struct {
-	// Version of Xdb to be deployed.
+	// Version of MongoDB to be deployed.
```
Comment (aerokite) [inline apis/kubedb/v1alpha1/mysql_types.go]: Incorrect comments

File: apis/kubedb/v1alpha1/mysql_types.go [hunk 4]
```diff
@@ -74,6 +74,6 @@ type MySQLStatus struct {
 type MySQLList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	// Items is a list of Xdb TPR objects
+	// Items is a list of MongoDB TPR objects
```
Comment (aerokite) [inline apis/kubedb/v1alpha1/mysql_types.go]: Do not use pointer in List Items. See Postgres & Elasticsearch

File: apis/kubedb/mongodb_types.go (hunk 1 above)
Comment (tamalsaha) [reply apis/kubedb/mongodb_types.go]: Use `mg`

File: apis/kubedb/v1alpha1/mysql_types.go (hunk 3 above)
Comment (the-redback) [reply apis/kubedb/v1alpha1/mysql_types.go]: Fixed in #145

File: apis/kubedb/v1alpha1/mysql_types.go (hunk 4 above)
Comment (the-redback) [reply apis/kubedb/v1alpha1/mysql_types.go]: Fixed in #145

**Comment by tamalsaha** [comment]
LGTM.

**Pull request merged**